lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -179,6 +179,16 @@
           }
         }
       }
+    } catch (final IllegalStateException ex) {
+      final RuntimeException transportException =
+                new TransportRuntimeException("tcpPortProvider failed to return free ports.");
+      LOG.log(Level.SEVERE, "tcpPortProvider failed to return free ports.");
```

Comment:
Please log the exception instead. Also, it might be good to add the range of ports tried to this exception message.